lib/Serverless.js

```diff
@@ -183,6 +183,26 @@ class Serverless {
     return this.version;
   }
 
+  // All uses of the `.serverless` folder should point to either the package
+  // output directory (getPackageOutputDir), or to the working directory
+  // (getWorkingTempDir). The framework and plugins should not assume that the
+  // `.serverless` folder will be used as the behaviour may change in the future
+  getWorkingTempDir() {
+    const serverlessTmpDirPath = path.join(this.config.servicePath || '.', '.serverless');
+    return serverlessTmpDirPath;
+  }
+  getPackageOutputDir() {
+    const packageOption =
+      this.processedInput && this.processedInput.options
+        ? this.processedInput.options.package
```

Comment:
`--package` option is unrelated to this work (it's handling should remain same, and to not over-complicate this PR I would not introduce any optimizations here, although they're welcome in any side PR, but still it'll be great to first discuss them)